pynsee/utils/init_conn.py

```diff
@@ -55,74 +51,69 @@ def init_conn(
-        "Local Data": "application/xml",
-    }
-
     invalid_requests = {}
 
-    user_agent = _get_requests_headers()
-
-    session = _get_requests_session()
-
-    for api, api_url in queries.items():
-        headers = {
-            "Accept": file_format[api],
-            "User-Agent": user_agent["User-Agent"],
-        }
-        if api == "Sirene":
-            headers["X-INSEE-Api-Key-Integration"] = sirene_key
-
-        results = session.get(
-            api_url, proxies=proxies, headers=headers, verify=False
-        )
-
-        code = results.status_code
-
-        if code == 429:
-            time.sleep(10)
-
-            results = requests.get(
-                api_url, proxies=proxies, headers=headers, verify=False
-            )
-
-        if results.status_code == 404:
-            RuntimeError(
-                f"Could not reach {api} at {api_url}, please get in touch if "
-                "the issue persists."
-            )
-        elif results.status_code != 200:
-            logger.critical(
-                f"Please subscribe to {api} API on api.insee.fr !\n"
-                f"Received error {results.status_code}: "
-            )
-
-            invalid_requests[api] = results.status_code
-
-    session.close()
+    with PynseeAPISession(
+        http_proxy=http_proxy, https_proxy=https_proxy, sirene_key=sirene_key
+    ) as session:
+
+        for api, api_url in queries.items():
+
+            try:
+                if api == "Sirene" and not session.sirene_key:
+                    # the user is probably not expecting to use SIRENE anyway:
+                    # simple warning and jump to next API in order to avoid
+                    # urllib retries
+                    logger.warning(
+                        f"Remember to subscribe to {api} API on api.insee.fr "
+                        "if you ever want to use it (type `help(init_conn)` "
+                        "to know more about this)"
+                    )
+                    invalid_requests[api] = 999
+
+                results = session.get(api_url, verify=False)
+            except requests.exceptions.RequestException as exc:
+                try:
+                    exc.response.status_code
+                except AttributeError:
+                    raise RuntimeError(
+                        f"Could not reach {api} at {api_url}, please control "
+                        "your proxy configuration."
+                    )
+                if exc.response.status_code == 404:
+                    raise RuntimeError(
+                        f"Could not reach {api} at {api_url}, please get in "
+                        "touch if the issue persists."
+                    )
+                elif results.status_code != 200:
+                    logger.critical(
+                        f"Please subscribe to {api} API on api.insee.fr !\n"
+                        f"Received error {exc.response.status_code}: "
+                    )
```

Comment:
so should this only happen with SIRENE? I'm not sure this is properly handled at the moment in terms of what we tell the person who will see the message...

Reply:
I think so, and I agree with your assessment about the warnings. I'm not sure either what would be the best course and I thought that could be kept for later...